File: jung-samples/src/main/java/edu/uci/ics/jung/samples/util/ControlHelpers.java
package edu.uci.ics.jung.samples.util;

import edu.uci.ics.jung.visualization.VisualizationServer;
import edu.uci.ics.jung.visualization.VisualizationViewer;
import edu.uci.ics.jung.visualization.control.CrossoverScalingControl;
import edu.uci.ics.jung.visualization.control.DefaultModalGraphMouse;
import edu.uci.ics.jung.visualization.control.ScalingControl;
import java.awt.*;
import javax.swing.*;

/**
 * @author Tom Nelson
 */
public class ControlHelpers {

  public static JComponent getZoomControls(VisualizationServer vv, String title) {

    final ScalingControl scaler = new CrossoverScalingControl();
    JButton plus = new JButton("+");
    plus.addActionListener(e -> scaler.scale(vv, 1.1f, vv.getCenter()));
    JButton minus = new JButton("-");
    minus.addActionListener(e -> scaler.scale(vv, 1 / 1.1f, vv.getCenter()));

    JPanel zoomPanel = new JPanel();
    zoomPanel.setBorder(BorderFactory.createTitledBorder(title));
    zoomPanel.add(plus);
    zoomPanel.add(minus);

    return zoomPanel;
  }

  public static JComponent getZoomControls(
      VisualizationServer vv, String title, LayoutManager buttonContainerLayoutManager) {

    final ScalingControl scaler = new CrossoverScalingControl();
    JButton plus = new JButton("+");
    plus.addActionListener(e -> scaler.scale(vv, 1.1f, vv.getCenter()));
    JButton minus = new JButton("-");
    minus.addActionListener(e -> scaler.scale(vv, 1 / 1.1f, vv.getCenter()));

    JPanel zoomPanel = new JPanel(buttonContainerLayoutManager);
    zoomPanel.setBorder(BorderFactory.createTitledBorder(title));
    zoomPanel.add(plus);
    zoomPanel.add(minus);

    return zoomPanel;
  }

  public static JComponent getModeControls(VisualizationViewer vv, String title) {
    final DefaultModalGraphMouse<Integer, Number> graphMouse =
        new DefaultModalGraphMouse<Integer, Number>();
    vv.setGraphMouse(graphMouse);

    JPanel modePanel = new JPanel(new GridLayout(2, 1));
    modePanel.setBorder(BorderFactory.createTitledBorder(title));
    modePanel.add(graphMouse.getModeComboBox());
    return modePanel;
  }
}

File: jung-samples/src/main/kotlin/edu/uci/ics/jung/samples/util/ControlHelpers.kt
package edu.uci.ics.jung.samples.util

import edu.uci.ics.jung.visualization.VisualizationServer
import edu.uci.ics.jung.visualization.VisualizationViewer
import edu.uci.ics.jung.visualization.control.CrossoverScalingControl
import edu.uci.ics.jung.visualization.control.DefaultModalGraphMouse
import java.awt.GridLayout
import java.awt.LayoutManager
import javax.swing.BorderFactory
import javax.swing.JButton
import javax.swing.JComponent
import javax.swing.JPanel

/**
 * @author Tom Nelson
 */
object ControlHelpers {

    @JvmStatic
    fun getZoomControls(vv: VisualizationServer<*, *>, title: String): JComponent {
        val scaler = CrossoverScalingControl()
        val plus = JButton("+")
        plus.addActionListener { scaler.scale(vv, 1.1f, vv.getCenter()) }
        val minus = JButton("-")
        minus.addActionListener { scaler.scale(vv, 1.0f / 1.1f, vv.getCenter()) }

        val zoomPanel = JPanel()
        zoomPanel.border = BorderFactory.createTitledBorder(title)
        zoomPanel.add(plus)
        zoomPanel.add(minus)

        return zoomPanel
    }

    @JvmStatic
    fun getZoomControls(
        vv: VisualizationServer<*, *>,
        title: String,
        buttonContainerLayoutManager: LayoutManager
    ): JComponent {
        val scaler = CrossoverScalingControl()
        val plus = JButton("+")
        plus.addActionListener { scaler.scale(vv, 1.1f, vv.getCenter()) }
        val minus = JButton("-")
        minus.addActionListener { scaler.scale(vv, 1.0f / 1.1f, vv.getCenter()) }

        val zoomPanel = JPanel(buttonContainerLayoutManager)
        zoomPanel.border = BorderFactory.createTitledBorder(title)
        zoomPanel.add(plus)
        zoomPanel.add(minus)

        return zoomPanel
    }

    @JvmStatic
    fun getModeControls(vv: VisualizationViewer<*, *>, title: String): JComponent {
        val graphMouse = DefaultModalGraphMouse<Int, Number>()
        vv.setGraphMouse(graphMouse)

        val modePanel = JPanel(GridLayout(2, 1))
        modePanel.border = BorderFactory.createTitledBorder(title)
        modePanel.add(graphMouse.getModeComboBox())
        return modePanel
    }
}
